test/cpp/microbenchmarks/bm_alarm.cc

```diff
@@ -0,0 +1,64 @@
+/*
+ *
+ * Copyright 2015 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+/* This benchmark exists to ensure that immediately-firing alarms are fast */
```

Comment:
I'm curious how you establish a baseline here. Fast in comparison to what?

Reply:
Great question. There's no baseline since we've never tested it before. I want this in here to prevent a regression in the future since I've just recommended it to an outside user on grpc-io as a way of injecting events into the CQ (which we knew already that other people were doing). The current speed on my Mac laptop was something like 311 ns so that's not a bad start IMO. (and I assume that it was faster in the actual test environment)

Reply:
SG